mlir/test/Transforms/lab4/kalinin_alexandr/test_kalinin.mlir

```diff
@@ -0,0 +1,77 @@
+    %2 = llvm.fmul %arg0, %0  : f64
+    %3 = llvm.fmul %arg1, %1  : f64
+    %4 = llvm.fadd %2, %3  : f64
+    llvm.return %4 : f64
+    // CHECK: %2 = llvm.fmul %arg1, %1  : f64
+    // CHECK-NEXT: %3 = llvm.intr.fma(%arg0, %0, %2)  : (f64, f64, f64) -> f64
+    // CHECK-NEXT: llvm.return %3 : f64
+  }
+
+  llvm.func local_unnamed_addr @_Z5func2dd(%arg0: f64 {llvm.noundef}, %arg1: f64 {llvm.noundef}) -> (f64 {llvm.noundef}) attributes {memory = #llvm.memory_effects<other = none, argMem = none, inaccessibleMem = none>, passthrough = ["mustprogress", "nofree", "norecurse", "nosync", "nounwind", "willreturn", ["uwtable", "2"], ["min-legal-vector-width", "0"], ["no-trapping-math", "true"], ["stack-protector-buffer-size", "8"], ["target-cpu", "x86-64"], ["target-features", "+cmov,+cx8,+fxsr,+mmx,+sse,+sse2,+x87"], ["tune-cpu", "generic"]]} {
+    %0 = llvm.mlir.constant(5.000000e+00 : f64) : f64
+    %1 = llvm.mlir.constant(2.000000e+00 : f64) : f64
+    %2 = llvm.fmul %arg0, %arg1  : f64
+    %3 = llvm.fadd %2, %0  : f64
+    %4 = llvm.fmul %2, %1  : f64
+    %5 = llvm.fadd %4, %3  : f64
+    llvm.return %5 : f64
+    // CHECK: %3 = llvm.intr.fma(%arg0, %arg1, %0)  : (f64, f64, f64) -> f64
+    // CHEK-NEXT: %4 = llvm.intr.fma(%2, %1, %3)  : (f64, f64, f64) -> f64
+    // CHEK-NEXT: llvm.return %5 : f64
+  }
+
+  llvm.func local_unnamed_addr @_Z5func3dd(%arg0: f64 {llvm.noundef}, %arg1: f64 {llvm.noundef}) -> (f64 {llvm.noundef}) attributes {memory = #llvm.memory_effects<other = none, argMem = none, inaccessibleMem = none>, passthrough = ["mustprogress", "nofree", "norecurse", "nosync", "nounwind", "willreturn", ["uwtable", "2"], ["min-legal-vector-width", "0"], ["no-trapping-math", "true"], ["stack-protector-buffer-size", "8"], ["target-cpu", "x86-64"], ["target-features", "+cmov,+cx8,+fxsr,+mmx,+sse,+sse2,+x87"], ["tune-cpu", "generic"]]} {
+    %0 = llvm.mlir.constant(2.000000e+00 : f64) : f64
+    %1 = llvm.mlir.constant(-7.000000e+00 : f64) : f64
+    %2 = llvm.fmul %arg0, %0  : f64
+    %3 = llvm.fadd %2, %1  : f64
+    llvm.return %3 : f64
+    // CHECK: %2 = llvm.intr.fma(%arg0, %0, %1)  : (f64, f64, f64) -> f64
+    // CHECK-NEXT: llvm.return %2 : f64
+  }
+
+  llvm.func local_unnamed_addr @_Z5func5ddd(%arg0: f64 {llvm.noundef}, %arg1: f64 {llvm.noundef}, %arg2: f64 {llvm.noundef}) -> (f64 {llvm.noundef}) attributes {memory = #llvm.memory_effects<other = none, argMem = none, inaccessibleMem = none>, passthrough = ["mustprogress", "nofree", "norecurse", "nosync", "nounwind", "willreturn", ["uwtable", "2"], ["min-legal-vector-width", "0"], ["no-trapping-math", "true"], ["stack-protector-buffer-size", "8"], ["target-cpu", "x86-64"], ["target-features", "+cmov,+cx8,+fxsr,+mmx,+sse,+sse2,+x87"], ["tune-cpu", "generic"]]} {
+    %0 = llvm.fmul %arg0, %arg1 : f64
+    %1 = llvm.fadd %0, %0 : f64
+    llvm.return %1 : f64
+    // CHECK: %0 = llvm.intr.fma(%arg0, %arg1, %arg0) : (f64, f64, f64) -> f64
+    // CHECK-NEXT: llvm.return %0 : f64
```

Comment:
This fusion is incorrect. 
```
%1 = llvm.fadd %0, %0 : f64
```
will not give this fma because the 3rd operand should be a result of multiplication, not just %arg0

Reply:
What's wrong? The result of the multiplication is written to %0 and then the result is added

Reply:
Problem fixed